src/bluesearch/database/article.py

```diff
@@ -73,6 +76,121 @@ def paragraphs(self) -> Iterable[Tuple[str, str]]:
         """
 
 
+class PubmedXMLParser(ArticleParser):
+    """Parser for PubMed XML files using the JATS Journal Publishing DTD.
+
+    Parameters
+    ----------
+    path
+        The path to the XML file from PubMed.
+    """
+
+    def __init__(self, path: Union[str, Path]) -> None:
+        self.content = etree.parse(str(path))  # nosec
+
+    @property
+    def title(self) -> str:
+        """Get the article title.
+
+        Returns
+        -------
+        str
+            The article title.
+        """
+        titles = self.content.find("//title-group/article-title")
+        return self.text_content(titles)
+
+    @property
+    def authors(self) -> Generator[str, None, None]:
+        """Get all author names.
+
+        Yields
+        ------
+        str
+            Every author, in the format "Given_Name(s) Surname".
+        """
+        authors = self.content.xpath('//contrib-group/contrib[@contrib-type="author"]')
+        for author in authors:
+            try:
+                given_names = self.text_content(author.find("name/given-names"))
+                surname = self.text_content(author.find("name/surname"))
+                author_str = given_names + " " + surname
+                yield author_str.strip()
+            except AttributeError:
+                # In rare cases, an author may not have a given name or a surname,
+                # e.g. it could be an organization. We decide to skip those.
+                continue
```

Comment:
Just wondering - is it the `author.find(...)` that can throw the `AttributeError`? Seems kind of strange.

Reply:
Done in f8b21a0. I forgot to update that part of the code after refactoring `text_content()`